tests/middleware/tests.py

```diff
@@ -301,7 +314,7 @@ def test_non_ascii_query_string_does_not_crash(self):
         request = self.rf.get('/slash')
         request.META['QUERY_STRING'] = force_str('drink=café')
         r = CommonMiddleware().process_request(request)
-        self.assertIsNone(r)
+        self.assertEqual(r.status_code, 301)
```

Comment:
Since `process_request()` returns a response, the second part of the test isn't needed (see 434d309ef6dbecbfd2b322d3a1da78aa5cb05fa8)